### PR TITLE
Add cdhitdup_cluster_sizes.tsv to DAG generation

### PR DESCRIPTION
**Pull request by gregdingle**
# Description

This is intended to replace the hacks in https://github.com/chanzuckerberg/idseq-dag/pull/252/files once the new read count logic has been tested. This PR adds `cdhitdup_cluster_sizes.tsv` as output to steps according to the following logic (copied from https://github.com/chanzuckerberg/idseq-dag/pull/252/files):

```
        if dag["name"] in ("postprocess", "non_host_alignment"):
            dag["targets"][cdhitdup_cluster_sizes_target] = [
                cdhitdup_cluster_sizes_path
            ]
            dag["given_targets"][cdhitdup_cluster_sizes_target] = {
                "s3_dir": dag["given_targets"]["host_filter_out"]["s3_dir"]
            }
        if dag["name"] == "host_filter":
            for fff in ["dedup1.fa.clstr", cdhitdup_cluster_sizes_path]:
                if fff not in dag["targets"]["cdhitdup_out"]:
                    dag["targets"]["cdhitdup_out"].append(fff)
```

And it adds `cdhitdup_cluster_sizes` as input according to:
```
            if s["class"] in ("PipelineStepBlastContigs", "PipelineStepRunAlignmentRemotely", "PipelineStepRunAssembly", "PipelineStepRunLZW", "PipelineStepRunBowtie2", "PipelineStepRunGsnapFilter", "PipelineStepRunSubsample", "PipelineStepGenerateAnnotatedFasta"):
                if cdhitdup_cluster_sizes_target not in s["in"]:
                    s["in"].append(cdhitdup_cluster_sizes_target)
```

# Notes

* **This will only be deployed once idseq-dag is ready for it.**
* I extended the dag extraction script by @kislyuk because it looked like the best way to see the changes. If the changes interfere with the purpose of the script then I'd be happy to remove them. 

# Tests

* Run extract script and inspect output. See JSON below. (Each step JSON was manually put in a JSON array.)

```
[
  {
    "name": "host_filter",
    "output_dir_s3": "s3://idseq-samples-development/samples/333/13444/results",
    "targets": {
      "fastqs": [
        "RR004_water_2_S23A_R1_001.fastq"
      ],
      "validate_input_out": [
        "validate_input_summary.json",
        "valid_input1.fastq"
      ],
      "star_out": [
        "unmapped1.fastq"
      ],
      "trimmomatic_out": [
        "trimmomatic1.fastq"
      ],
      "priceseq_out": [
        "priceseq1.fa"
      ],
      "cdhitdup_out": [
        "dedup1.fa",
        "dedup1.fa.clstr",
        "cdhitdup_cluster_sizes.tsv"
      ],
      "lzw_out": [
        "lzw1.fa"
      ],
      "bowtie2_out": [
        "bowtie2_1.fa"
      ],
      "subsampled_out": [
        "subsampled_1.fa"
      ],
      "gsnap_filter_out": [
        "gsnap_filter_1.fa"
      ]
    },
    "steps": [
      {
        "in": [
          "fastqs"
        ],
        "out": "validate_input_out",
        "class": "PipelineStepRunValidateInput",
        "module": "idseq_dag.steps.run_validate_input",
        "additional_files": {},
        "additional_attributes": {
          "truncate_fragments_to": 75000000,
          "file_ext": "fastq"
        }
      },
      {
        "in": [
          "validate_input_out"
        ],
        "out": "star_out",
        "class": "PipelineStepRunStar",
        "module": "idseq_dag.steps.run_star",
        "additional_files": {
          "star_genome": "s3://idseq-database/host_filter/human/2018-02-15-utc-1518652800-unixtime__2018-02-15-utc-1518652800-unixtime/human_STAR_genome.tar"
        },
        "additional_attributes": {
          "output_gene_file": "reads_per_gene.star.tab",
          "nucleotide_type": "DNA",
          "host_genome": "human",
          "output_metrics_file": "picard_insert_metrics.txt",
          "output_histogram_file": "insert_size_histogram.pdf"
        }
      },
      {
        "in": [
          "star_out"
        ],
        "out": "trimmomatic_out",
        "class": "PipelineStepRunTrimmomatic",
        "module": "idseq_dag.steps.run_trimmomatic",
        "additional_files": {
          "adapter_fasta": "s3://idseq-database/adapter_sequences/illumina_TruSeq3-SE.fasta"
        },
        "additional_attributes": {}
      },
      {
        "in": [
          "trimmomatic_out"
        ],
        "out": "priceseq_out",
        "class": "PipelineStepRunPriceSeq",
        "module": "idseq_dag.steps.run_priceseq",
        "additional_files": {},
        "additional_attributes": {}
      },
      {
        "in": [
          "priceseq_out"
        ],
        "out": "cdhitdup_out",
        "class": "PipelineStepRunCDHitDup",
        "module": "idseq_dag.steps.run_cdhitdup",
        "additional_files": {},
        "additional_attributes": {}
      },
      {
        "in": [
          "cdhitdup_out",
          "cdhitdup_cluster_sizes"
        ],
        "out": "lzw_out",
        "class": "PipelineStepRunLZW",
        "module": "idseq_dag.steps.run_lzw",
        "additional_files": {},
        "additional_attributes": {
          "thresholds": [
            0.45,
            0.42
          ],
          "threshold_readlength": 150
        }
      },
      {
        "in": [
          "lzw_out",
          "cdhitdup_cluster_sizes"
        ],
        "out": "bowtie2_out",
        "class": "PipelineStepRunBowtie2",
        "module": "idseq_dag.steps.run_bowtie2",
        "additional_files": {
          "bowtie2_genome": "s3://idseq-database/host_filter/human/2018-02-15-utc-1518652800-unixtime__2018-02-15-utc-1518652800-unixtime/human_bowtie2_genome.tar"
        },
        "additional_attributes": {
          "output_sam_file": "bowtie2.sam"
        }
      },
      {
        "in": [
          "bowtie2_out",
          "cdhitdup_cluster_sizes"
        ],
        "out": "subsampled_out",
        "class": "PipelineStepRunSubsample",
        "module": "idseq_dag.steps.run_subsample",
        "additional_files": {},
        "additional_attributes": {
          "max_fragments": 1000000
        }
      },
      {
        "in": [
          "subsampled_out",
          "cdhitdup_cluster_sizes"
        ],
        "out": "gsnap_filter_out",
        "class": "PipelineStepRunGsnapFilter",
        "module": "idseq_dag.steps.run_gsnap_filter",
        "additional_files": {
          "gsnap_genome": "s3://idseq-database/host_filter/human/2018-02-15-utc-1518652800-unixtime__2018-02-15-utc-1518652800-unixtime/hg38_pantro5_k16.tar"
        },
        "additional_attributes": {
          "output_sam_file": "gsnap_filter.sam"
        }
      }
    ],
    "given_targets": {
      "fastqs": {
        "s3_dir": "s3://idseq-samples-development/samples/333/13444/fastqs",
        "count_reads": 1,
        "max_fragments": 75000000
      }
    }
  },
  {
    "name": "non_host_alignment",
    "output_dir_s3": "s3://idseq-samples-development/samples/333/13444/results",
    "targets": {
      "host_filter_out": [
        "gsnap_filter_1.fa"
      ],
      "gsnap_out": [
        "gsnap.m8",
        "gsnap.deduped.m8",
        "gsnap.hitsummary.tab",
        "gsnap_counts.json"
      ],
      "rapsearch2_out": [
        "rapsearch2.m8",
        "rapsearch2.deduped.m8",
        "rapsearch2.hitsummary.tab",
        "rapsearch2_counts.json"
      ],
      "taxon_count_out": [
        "taxon_counts.json"
      ],
      "annotated_out": [
        "annotated_merged.fa",
        "unidentified.fa"
      ],
      "cdhitdup_cluster_sizes": [
        "cdhitdup_cluster_sizes.tsv"
      ]
    },
    "steps": [
      {
        "in": [
          "host_filter_out",
          "cdhitdup_cluster_sizes"
        ],
        "out": "gsnap_out",
        "class": "PipelineStepRunAlignmentRemotely",
        "module": "idseq_dag.steps.run_alignment_remotely",
        "additional_files": {
          "lineage_db": "s3://idseq-database/taxonomy/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/taxid-lineages.db",
          "accession2taxid_db": "s3://idseq-database/alignment_data/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/accession2taxid.db",
          "deuterostome_db": "s3://idseq-database/taxonomy/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/deuterostome_taxids.txt"
        },
        "additional_attributes": {
          "service": "gsnap",
          "chunks_in_flight": 32,
          "chunk_size": 60000,
          "max_concurrent": 2,
          "environment": "staging",
          "max_interval_between_describe_instances": 900,
          "job_tag_prefix": "RunningIDseqBatchJob_",
          "job_tag_refresh_seconds": 600,
          "draining_tag": "draining",
          "index_dir_suffix": "2018-04-01"
        }
      },
      {
        "in": [
          "host_filter_out",
          "cdhitdup_cluster_sizes"
        ],
        "out": "rapsearch2_out",
        "class": "PipelineStepRunAlignmentRemotely",
        "module": "idseq_dag.steps.run_alignment_remotely",
        "additional_files": {
          "lineage_db": "s3://idseq-database/taxonomy/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/taxid-lineages.db",
          "accession2taxid_db": "s3://idseq-database/alignment_data/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/accession2taxid.db",
          "deuterostome_db": "s3://idseq-database/taxonomy/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/deuterostome_taxids.txt"
        },
        "additional_attributes": {
          "service": "rapsearch2",
          "chunks_in_flight": 32,
          "chunk_size": 80000,
          "max_concurrent": 8,
          "environment": "staging",
          "max_interval_between_describe_instances": 900,
          "job_tag_prefix": "RunningIDseqBatchJob_",
          "job_tag_refresh_seconds": 600,
          "draining_tag": "draining",
          "index_dir_suffix": "2018-04-01"
        }
      },
      {
        "in": [
          "gsnap_out",
          "rapsearch2_out"
        ],
        "out": "taxon_count_out",
        "class": "PipelineStepCombineTaxonCounts",
        "module": "idseq_dag.steps.combine_taxon_counts",
        "additional_files": {},
        "additional_attributes": {}
      },
      {
        "in": [
          "host_filter_out",
          "gsnap_out",
          "rapsearch2_out",
          "cdhitdup_cluster_sizes"
        ],
        "out": "annotated_out",
        "class": "PipelineStepGenerateAnnotatedFasta",
        "module": "idseq_dag.steps.generate_annotated_fasta",
        "additional_files": {},
        "additional_attributes": {}
      }
    ],
    "given_targets": {
      "host_filter_out": {
        "s3_dir": "s3://idseq-samples-development/samples/333/13444/results/"
      },
      "cdhitdup_cluster_sizes": {
        "s3_dir": "s3://idseq-samples-development/samples/333/13444/results/"
      }
    }
  },
  {
    "name": "postprocess",
    "output_dir_s3": "s3://idseq-samples-development/samples/333/13444/postprocess",
    "targets": {
      "host_filter_out": [
        "gsnap_filter_1.fa"
      ],
      "gsnap_out": [
        "gsnap.m8",
        "gsnap.deduped.m8",
        "gsnap.hitsummary.tab",
        "gsnap_counts.json"
      ],
      "rapsearch2_out": [
        "rapsearch2.m8",
        "rapsearch2.deduped.m8",
        "rapsearch2.hitsummary.tab",
        "rapsearch2_counts.json"
      ],
      "assembly_out": [
        "assembly/contigs.fasta",
        "assembly/scaffolds.fasta",
        "assembly/read-contig.sam",
        "assembly/contig_stats.json"
      ],
      "coverage_out": [
        "assembly/contig_coverage.json",
        "assembly/contig_coverage_summary.csv"
      ],
      "gsnap_accessions_out": [
        "assembly/nt.refseq.fasta"
      ],
      "rapsearch2_accessions_out": [
        "assembly/nr.refseq.fasta"
      ],
      "refined_gsnap_out": [
        "assembly/gsnap.blast.m8",
        "assembly/gsnap.reassigned.m8",
        "assembly/gsnap.hitsummary2.tab",
        "assembly/refined_gsnap_counts.json",
        "assembly/gsnap_contig_summary.json",
        "assembly/gsnap.blast.top.m8"
      ],
      "refined_rapsearch2_out": [
        "assembly/rapsearch2.blast.m8",
        "assembly/rapsearch2.reassigned.m8",
        "assembly/rapsearch2.hitsummary2.tab",
        "assembly/refined_rapsearch2_counts.json",
        "assembly/rapsearch2_contig_summary.json",
        "assembly/rapsearch2.blast.top.m8"
      ],
      "refined_taxon_count_out": [
        "assembly/refined_taxon_counts.json"
      ],
      "contig_summary_out": [
        "assembly/combined_contig_summary.json"
      ],
      "refined_annotated_out": [
        "assembly/refined_annotated_merged.fa",
        "assembly/refined_unidentified.fa"
      ],
      "refined_taxid_fasta_out": [
        "assembly/refined_taxid_annot.fasta"
      ],
      "refined_taxid_locator_out": [
        "assembly/refined_taxid_annot_sorted_nt.fasta",
        "assembly/refined_taxid_locations_nt.json",
        "assembly/refined_taxid_annot_sorted_nr.fasta",
        "assembly/refined_taxid_locations_nr.json",
        "assembly/refined_taxid_annot_sorted_genus_nt.fasta",
        "assembly/refined_taxid_locations_genus_nt.json",
        "assembly/refined_taxid_annot_sorted_genus_nr.fasta",
        "assembly/refined_taxid_locations_genus_nr.json",
        "assembly/refined_taxid_annot_sorted_family_nt.fasta",
        "assembly/refined_taxid_locations_family_nt.json",
        "assembly/refined_taxid_annot_sorted_family_nr.fasta",
        "assembly/refined_taxid_locations_family_nr.json",
        "assembly/refined_taxid_locations_combined.json"
      ],
      "cdhitdup_cluster_sizes": [
        "cdhitdup_cluster_sizes.tsv"
      ]
    },
    "steps": [
      {
        "in": [
          "host_filter_out",
          "cdhitdup_cluster_sizes"
        ],
        "out": "assembly_out",
        "class": "PipelineStepRunAssembly",
        "module": "idseq_dag.steps.run_assembly",
        "additional_files": {},
        "additional_attributes": {
          "memory": 200
        }
      },
      {
        "in": [
          "assembly_out"
        ],
        "out": "coverage_out",
        "class": "PipelineStepGenerateCoverageStats",
        "module": "idseq_dag.steps.generate_coverage_stats",
        "additional_files": {},
        "additional_attributes": {}
      },
      {
        "in": [
          "gsnap_out"
        ],
        "out": "gsnap_accessions_out",
        "class": "PipelineStepDownloadAccessions",
        "module": "idseq_dag.steps.download_accessions",
        "additional_files": {
          "lineage_db": "s3://idseq-database/taxonomy/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/taxid-lineages.db",
          "loc_db": "s3://idseq-database/alignment_data/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/nt_loc.db",
          "db": "s3://idseq-database/alignment_data/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/nt"
        },
        "additional_attributes": {
          "db": "s3://idseq-database/alignment_data/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/nt",
          "db_type": "nt"
        }
      },
      {
        "in": [
          "rapsearch2_out"
        ],
        "out": "rapsearch2_accessions_out",
        "class": "PipelineStepDownloadAccessions",
        "module": "idseq_dag.steps.download_accessions",
        "additional_files": {
          "lineage_db": "s3://idseq-database/taxonomy/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/taxid-lineages.db",
          "loc_db": "s3://idseq-database/alignment_data/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/nr_loc.db",
          "db": "s3://idseq-database/alignment_data/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/nr"
        },
        "additional_attributes": {
          "db": "s3://idseq-database/alignment_data/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/nr",
          "db_type": "nr"
        }
      },
      {
        "in": [
          "gsnap_out",
          "assembly_out",
          "gsnap_accessions_out",
          "cdhitdup_cluster_sizes"
        ],
        "out": "refined_gsnap_out",
        "class": "PipelineStepBlastContigs",
        "module": "idseq_dag.steps.blast_contigs",
        "additional_files": {
          "lineage_db": "s3://idseq-database/taxonomy/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/taxid-lineages.db",
          "deuterostome_db": "s3://idseq-database/taxonomy/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/deuterostome_taxids.txt"
        },
        "additional_attributes": {
          "db_type": "nt"
        }
      },
      {
        "in": [
          "rapsearch2_out",
          "assembly_out",
          "rapsearch2_accessions_out",
          "cdhitdup_cluster_sizes.tsv"
        ],
        "out": "refined_rapsearch2_out",
        "class": "PipelineStepBlastContigs",
        "module": "idseq_dag.steps.blast_contigs",
        "additional_files": {
          "lineage_db": "s3://idseq-database/taxonomy/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/taxid-lineages.db",
          "deuterostome_db": "s3://idseq-database/taxonomy/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/deuterostome_taxids.txt"
        },
        "additional_attributes": {
          "db_type": "nr"
        }
      },
      {
        "in": [
          "refined_gsnap_out",
          "refined_rapsearch2_out"
        ],
        "out": "refined_taxon_count_out",
        "class": "PipelineStepCombineTaxonCounts",
        "module": "idseq_dag.steps.combine_taxon_counts",
        "additional_files": {},
        "additional_attributes": {}
      },
      {
        "in": [
          "refined_gsnap_out",
          "refined_rapsearch2_out"
        ],
        "out": "contig_summary_out",
        "class": "PipelineStepCombineJson",
        "module": "idseq_dag.steps.combine_json",
        "additional_files": {},
        "additional_attributes": {
          "field_idx": 4
        }
      },
      {
        "in": [
          "host_filter_out",
          "refined_gsnap_out",
          "refined_rapsearch2_out",
          "cdhitdup_cluster_sizes"
        ],
        "out": "refined_annotated_out",
        "class": "PipelineStepGenerateAnnotatedFasta",
        "module": "idseq_dag.steps.generate_annotated_fasta",
        "additional_files": {},
        "additional_attributes": {}
      },
      {
        "in": [
          "refined_annotated_out",
          "refined_gsnap_out",
          "refined_rapsearch2_out"
        ],
        "out": "refined_taxid_fasta_out",
        "class": "PipelineStepGenerateTaxidFasta",
        "module": "idseq_dag.steps.generate_taxid_fasta",
        "additional_files": {
          "lineage_db": "s3://idseq-database/taxonomy/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/taxid-lineages.db"
        },
        "additional_attributes": {}
      },
      {
        "in": [
          "refined_taxid_fasta_out"
        ],
        "out": "refined_taxid_locator_out",
        "class": "PipelineStepGenerateTaxidLocator",
        "module": "idseq_dag.steps.generate_taxid_locator",
        "additional_files": {},
        "additional_attributes": {}
      }
    ],
    "given_targets": {
      "host_filter_out": {
        "s3_dir": "s3://idseq-samples-development/samples/333/13444/results/"
      },
      "gsnap_out": {
        "s3_dir": "s3://idseq-samples-development/samples/333/13444/results/"
      },
      "rapsearch2_out": {
        "s3_dir": "s3://idseq-samples-development/samples/333/13444/results/"
      },
      "cdhitdup_cluster_sizes": {
        "s3_dir": "s3://idseq-samples-development/samples/333/13444/results/"
      }
    }
  },
  {
    "name": "experimental",
    "output_dir_s3": "s3://idseq-samples-development/samples/333/13444/postprocess",
    "targets": {
      "taxid_fasta_in": [
        "annotated_merged.fa",
        "gsnap.hitsummary.tab",
        "rapsearch2.hitsummary.tab"
      ],
      "refined_gsnap_in": [
        "gsnap.reassigned.m8",
        "gsnap.hitsummary2.tab",
        "gsnap.blast.top.m8"
      ],
      "contig_in": [
        "contig_coverage.json",
        "contig_stats.json",
        "contigs.fasta"
      ],
      "gsnap_m8": [
        "gsnap.deduped.m8"
      ],
      "taxid_fasta_out": [
        "taxid_annot.fasta"
      ],
      "taxid_locator_out": [
        "taxid_annot_sorted_nt.fasta",
        "taxid_locations_nt.json",
        "taxid_annot_sorted_nr.fasta",
        "taxid_locations_nr.json",
        "taxid_annot_sorted_genus_nt.fasta",
        "taxid_locations_genus_nt.json",
        "taxid_annot_sorted_genus_nr.fasta",
        "taxid_locations_genus_nr.json",
        "taxid_annot_sorted_family_nt.fasta",
        "taxid_locations_family_nt.json",
        "taxid_annot_sorted_family_nr.fasta",
        "taxid_locations_family_nr.json",
        "taxid_locations_combined.json"
      ],
      "alignment_viz_out": [
        "align_viz.summary"
      ],
      "srst2_out": [
        "out.log",
        "out__genes__ARGannot_r2__results.txt",
        "out__fullgenes__ARGannot_r2__results.txt",
        "amr_processed_results.csv",
        "amr_summary_results.csv",
        "output__.ARGannot_r2.sorted.bam"
      ],
      "coverage_viz_out": [
        "coverage_viz_summary.json"
      ],
      "fastqs": [
        "RR004_water_2_S23A_R1_001.fastq"
      ],
      "nonhost_fasta": [
        "refined_taxid_annot.fasta"
      ],
      "nonhost_fastq_out": [
        "nonhost_R1.fastq"
      ]
    },
    "steps": [
      {
        "in": [
          "taxid_fasta_in"
        ],
        "out": "taxid_fasta_out",
        "class": "PipelineStepGenerateTaxidFasta",
        "module": "idseq_dag.steps.generate_taxid_fasta",
        "additional_files": {
          "lineage_db": "s3://idseq-database/taxonomy/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/taxid-lineages.db"
        },
        "additional_attributes": {}
      },
      {
        "in": [
          "taxid_fasta_out"
        ],
        "out": "taxid_locator_out",
        "class": "PipelineStepGenerateTaxidLocator",
        "module": "idseq_dag.steps.generate_taxid_locator",
        "additional_files": {},
        "additional_attributes": {}
      },
      {
        "in": [
          "gsnap_m8",
          "taxid_locator_out"
        ],
        "out": "alignment_viz_out",
        "class": "PipelineStepGenerateAlignmentViz",
        "module": "idseq_dag.steps.generate_alignment_viz",
        "additional_files": {
          "nt_loc_db": "s3://idseq-database/alignment_data/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/nt_loc.db",
          "nt_db": "s3://idseq-database/alignment_data/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/nt"
        },
        "additional_attributes": {
          "nt_db": "s3://idseq-database/alignment_data/2018-04-01-utc-1522569777-unixtime__2018-04-04-utc-1522862260-unixtime/nt"
        }
      },
      {
        "in": [
          "fastqs"
        ],
        "out": "srst2_out",
        "class": "PipelineStepRunSRST2",
        "module": "idseq_dag.steps.run_srst2",
        "additional_files": {
          "resist_gene_db": "s3://idseq-database/amr/ARGannot_r2.fasta",
          "resist_genome_bed": "s3://idseq-database/amr/argannot_genome.bed"
        },
        "additional_attributes": {
          "min_cov": 0,
          "n_threads": 16,
          "file_ext": "fastq"
        }
      },
      {
        "in": [
          "refined_gsnap_in",
          "contig_in",
          "gsnap_m8"
        ],
        "out": "coverage_viz_out",
        "class": "PipelineStepGenerateCoverageViz",
        "module": "idseq_dag.steps.generate_coverage_viz",
        "additional_files": {
          "info_db": "s3://idseq-database/alignment_data/2020-02-10/nt_info.sqlite3"
        },
        "additional_attributes": {}
      },
      {
        "in": [
          "fastqs",
          "nonhost_fasta"
        ],
        "out": "nonhost_fastq_out",
        "class": "PipelineStepNonhostFastq",
        "module": "idseq_dag.steps.nonhost_fastq",
        "additional_files": {},
        "additional_attributes": {}
      }
    ],
    "given_targets": {
      "taxid_fasta_in": {
        "s3_dir": "s3://idseq-samples-development/samples/333/13444/results/"
      },
      "gsnap_m8": {
        "s3_dir": "s3://idseq-samples-development/samples/333/13444/results/"
      },
      "refined_gsnap_in": {
        "s3_dir": "s3://idseq-samples-development/samples/333/13444/postprocess//assembly"
      },
      "contig_in": {
        "s3_dir": "s3://idseq-samples-development/samples/333/13444/postprocess//assembly"
      },
      "fastqs": {
        "s3_dir": "s3://idseq-samples-development/samples/333/13444/fastqs"
      },
      "nonhost_fasta": {
        "s3_dir": "s3://idseq-samples-development/samples/333/13444/postprocess//assembly"
      }
    }
  }
]
```

### CODE REVIEW

File: lib/tasks/export_dags.rake [hunk 1]
```diff
@@ -15,16 +21,20 @@ task :export_dags, [:sample_id] => :environment do |_, args|
     end
   end
 
-  if args.sample_id?
+  if args.sample_id
```
Comment (gregdingle) [inline lib/tasks/export_dags.rake]: this line just didn't work for me until I changed it... `sample_id?` was always `nil`

File: lib/tasks/export_dags.rake [hunk 2]
```diff
@@ -1,10 +1,16 @@
-task :export_dags, [:sample_id] => :environment do |_, args|
+task :export_dags, [:sample_id, :print, :step] => :environment do |_, args|
```
Comment (cdebourcy) [inline lib/tasks/export_dags.rake]: I approved the PR based on the jbuilder files. I defer to @kislyuk about whether the changes to the export_dags.rake file are fine -- a priori I'd probably err on the side of only changing the dag jbuilder for this PR

**Comment by gregdingle** [comment]
Thanks @cdebourcy . @boris-dimitrov could you check as well? To re-iterate, I tested this change only by inspecting the output, not by running it through idseq-dag, which would required https://github.com/chanzuckerberg/idseq-dag/pull/252 to be finished.

**Comment by cdebourcy** [comment]
Let's get this change ready for deployment by doing the testing already. You can do the end-to-end testing change by submitting a sample from your local environment (and scaling up staging alignment machines). You can upload the sample from localhost:3000/samples/new (legacy interface) and specify a branch of idseq-dag to be used for processing, which should essentially be https://github.com/chanzuckerberg/idseq-dag/pull/252/files minus the hacks you're seeking to remove.

**Comment by boris-dimitrov** [comment]
I am taking this over for some small updates and also because we need to add the ability of the result loader to read taxon_counts_with_dcr.json and load the "count" field from there.  Currently if it tries to read that file, it breaks because there are extra fields it doesn't recognize.

**Comment by boris-dimitrov** [comment]
The test plan, btw, is:

1) Test this from local env with regular currently deployed idseq-dag.   It should work that way.   If not, fix either this PR or the deployed idseq-dag, as needed.

2) Test this from local env with idseq-dag branch that sets count.DAG_SURGERY_HACKS_FOR_READ_COUNTING=False.   If passes, merge this PR.  Don't make any changes to the deployed idseq-dag.

3) After this PR is deployed, and happy, remove from idseq-dag all the code that executes when DAG_SURGERY_HACKS_FOR_READ_COUNTING=True.

This PR is now superceded by #3189

**Comment by gregdingle** [comment]
Replaced by https://github.com/chanzuckerberg/idseq-web/pull/3189

**Pull request closed without merging**